array_family/flex_ext.py

```diff
@@ -606,6 +591,72 @@ def __init__(self):
         mask2.set_selected(sind.select(mask), True)
         return mask2, other_matched, other_unmatched
 
+    def match(
+        self,
+        other: dials_array_family_flex_ext.reflection_table,
+        max_separation: int = 2,
+        key: str = "xyzobs.px.value",
+        scale: Tuple[float, float, float] = (1, 1, 1),
+    ) -> Tuple[
+        cctbx.array_family.flex.size_t,
+        cctbx.array_family.flex.size_t,
+        cctbx.array_family.flex.double,
+    ]:
+        """
+        Match reflections from this reflection list to another (reference) list
+
+        The match is based on comparison of any chosen 3-vector column.
+
+        Args:
+            other: The reflection table to match against
+            max_separations: Maximum distance in column-space to match on
+            key: Name of 3-vector column
+            scale:
+                Relative scales to apply to vectors before matching in
+                case e.g. very wide or very fine slicing.
+
+        Returns:
+            (self_index, other_index, distance) where:
+                self_index: is the reindex in the this reflection table
+                other_index: is the index in the other reflection table
+                distance: The calculated distance of specified column-vector
+
+            such that self[self_index][j] ~= other[other_index][j].
+        """
+
+        xyz = self[key]
+        ref = other[key]
+
+        flex = cctbx.array_family.flex
```

Comment:
Please no? We should try to be consistent in the file, so either use `flex` everywhere else, or use `cctbx.array_family.flex` four times in here. I think the former is slightly too confusing (Reasonable to assume flex == dials.af.flex inside dials) so would prefer the latter as explicit. Maybe defining `import cctbx.array_family.flex as caf_flex` is a middle ground? (although it feels a little bit dirty)